### ISSUE TITLE
Pear down content in README to essentials

### ISSUE THREAD

**Issue by thrau** [localstack/localstack#5023]
The README has grown over the years as it was the main source of documentation for the community version of localstack. Now that we have https://docs.localstack.cloud, I think it's time to cut the README down to essential information.

I think we should keep the "Overview", "Install", "Run", "Play" sections linked in the header to a few sentences total, just enough for people to get started, and remove everything else that is documentation (link to the docs in the header should suffice).
Additionally, we may want to keep the "License", and "Contributing sections", and  I can see that we may want to keep the backers and announcements section. The contributors I think we could dispense with, since we have the list of contributors as a github feature.

thoughts @whummer ?

Here are some examples of other projects that have a minimal readme:
* django: https://github.com/django/django/blob/main/README.rst
* flask: https://github.com/pallets/flask/blob/main/README.rst
* kubernetes: https://github.com/kubernetes/kubernetes/blob/master/README.md
* Terraform: https://github.com/hashicorp/terraform/blob/main/README.md
* black: https://github.com/psf/black/blob/main/README.md
* moto: https://github.com/spulec/moto/blob/master/README.md

**Comment by championshuttler** [comment]
This is the issue I was about to file, as a new contributor readme is overwhelming at the moment, I can work on this issue

**Comment by thrau** [comment]
hey @championshuttler thanks for the feebdack. can you share some READMEs of projects that you like?

**Comment by championshuttler** [comment]
Hey @thrau, https://github.com/mozilla/addons-frontend/ this is one of the project I used to work on, Currently its little bit extended but our api specification and other documents we host them https://addons-server.readthedocs.io/en/latest/index.html so we could do pretty much same here as well, just keep the introduction, contribution, install, run, licensing and other necessary information. Readme is the key to invite contributors and increase community IMO and atm its bit overwhelming at first look

**Comment by HarshCasper** [comment]
Hi @thrau @championshuttler 

We can keep only the following information on the README:

- Overview
- Installation & Usage
- Resources
- Contributing
- LICENSE

We can remove the `Contributors` section as you mentioned since it's a GitHub feature in itself. We can dilute the areas around troubleshooting, configuration, announcements, development in separate markdown files which we can link in the `Resources` section. 

The interaction, integration have already been documented on the main documentation website and hence it would be redundant to have them on the README. 

What do you think about it?

**Comment by whummer** [comment]
Thanks @HarshCasper @thrau @championshuttler - this restructuring will be very useful! We can keep only the essentials, and point to sub-READMEs or external pointers in our docs. 👍 

> We can remove the Contributors section as you mentioned since it's a GitHub feature in itself

The Contributors/Backers/Sponsors sections had been previously added when we got onboarded to [OpenCollective](https://opencollective.com). I think it's fine if we don't publish this section as prominently in the main README - we could include it in a sub-README instead (e.g., in `doc/contributors/...`).

Overall, your proposal sounds good to me. Should we start with a first draft PR in that direction @HarshCasper ? 👍

**Comment by thrau** [comment]
> The interaction, integration have already been documented on the main documentation website and hence it would be redundant to have them on the README.

personally i think a really short intro with a couple of commands to get started wouldn't hurt. basically just to demonstrate how easy it is to use:

```
pip install localstack awscli-local
localstack start -d
localstack wait
```

```
awslocal sqs create-queue --queue-name "my-queue"
awslocal sqs list-queues
```

but open to removing it completely.

**Comment by HarshCasper** [comment]
Fixed in #5100
